data/list/basic.lean

```diff
@@ -1470,6 +1491,24 @@ theorem infix_of_mem_join : ∀ {L : list (list α)} {l}, l ∈ L → l <:+: joi
 | (l' :: L) l (or.inr h)   :=
   is_infix.trans (infix_of_mem_join h) $ infix_of_suffix $ suffix_append _ _
 
+@[simp] lemma append_eq_nil {β} (p q : list β) : (p ++ q) = [] ↔ p = [] ∧ q = [] :=
+begin
+split,
+show (p ++ q) = [] → p = [] ∧ q = [],
+{
+  intro h,
+  split,
+  apply eq_nil_of_prefix_nil, rw ← h, simp, 
```

Comment:
add `{ .. }`

Reply:
maybe also `split; { apply eq_nil_of_prefix_nil, rw ← h, simp }` ?

Reply:
I added `{ ... }`. We can't use `;` since we're applying two different lemmas (note `prefix` vs `suffix` in the two cases.